modules/seqexec/engine/src/main/scala/seqexec/engine/Execution.scala

```diff
@@ -25,6 +26,13 @@ final case class Execution[F[_]](execution: List[Action[F]]) {
 
   val results: List[Action[F]] = execution.filter(Action.finished)
 
+  def toParallelActionsList: List[ParallelActions[F]] =
+    if (execution.isEmpty) {
+      Nil
+    } else {
+      List(NonEmptyList.fromListUnsafe(execution))
+    }
```

Comment:
```scala
def toParallelActionsList: List[ParallelActions[F]] = 
  NonEmptyList.fromList(execution).toList
```

Reply:
Good, I guess we should refrain to use `fromListUnsafe` even though we know the list is empty